### ISSUE TITLE
But this plugin creates 1 real page for each member instead of a template?

### ISSUE THREAD

**Issue by DavideWFP** [strangerstudios/pmpro-user-pages#11]
I will import 500 members, will i get 500 individual pages?

How am I supposed to style them individually one by one if i change something to the user profile page?

**Comment by dparker1005** [comment]
> I will import 500 members, will i get 500 individual pages?
Yes, as long as users are given a membership level during the import, a page will be created for each user along with a sub-page for each membership level that they have.

> How am I supposed to style them individually one by one if i change something to the user profile page?
It depends on what you mean by the "user profile page". If you are referring to the front-end profile added in PMPro 2.3, that page is not related to the PMPro User Pages add-on. This add-on should be used if you would like a separate page for each user that only you and that user can see, that can also be edited independent per-user.

You can edit the default page contents of each user's user page by using the `pmpro_user_page_postdata` filter: https://github.com/strangerstudios/pmpro-user-pages/blob/66c2d9ade7ab3c10888e821423cefcee81ec4511/pmpro-user-pages.php#L51

And you can edit the default page contents of the page that is created for each membership for each user using the `pmpro_user_page_purchase_postdata` filter: https://github.com/strangerstudios/pmpro-user-pages/blob/66c2d9ade7ab3c10888e821423cefcee81ec4511/pmpro-user-pages.php#L74

If you would like to bulk-update pages after they are already created, this would require custom code or a SQL query. Here is some code where we list all user pages for all users, which you can use as a starting point for bulk-updating if that is something that you would like to pursue: https://github.com/strangerstudios/pmpro-user-pages/blob/66c2d9ade7ab3c10888e821423cefcee81ec4511/pmpro-user-pages.php#L190-L200
